src/api/server.rkt

```diff
@@ -67,6 +117,13 @@
   (log "Getting timeline for job: ~a.\n" job-id)
   (place-channel-get a))
 
+; Returns #f if there is no job returns the job-id if there is a completed job.
+(define (server-check-on job-id)
+  (define-values (a b) (place-channel))
+  (place-channel-put manager (list 'check job-id b))
+  (log "Checking on: ~a.\n" job-id)
+  (place-channel-get a))
```

Comment:
I still think this `place-channel` / `place-channel-put` / `place-channel-get` pattern should be combined into a helper function, and then we wouldn't actually need this function at all (nor lots of others).